pkg/consts/consts.go

```diff
@@ -11,6 +11,23 @@ const (
 	// tx inclusion proof via the ABCI Query method. The desired transaction
 	// index must be formatted into the path.
 	TxInclusionProofQueryPath = "custom/txInclusionProof/%d"
+	// TxSharesQueryPath is the path used to query the application for the shares range
+	// where the transaction live via the ABCI query method.
+	TxSharesQueryPath = "custom/txShares/%d"
+	// MessageSharesQueryPath is the path used to query the application for the shares range
+	// where the message of a transaction live via the ABCI query method.
+	MessageSharesQueryPath = "custom/msgShares/%d"
+	// ShareInclusionProofQueryPath is the path used to query the application for the
+	// shares to rows NMT inclusion proof via the ABCI query method.
+	ShareInclusionProofQueryPath = "custom/shareInclusionProof/%d/%d"
+	// RowsInclusionProofQueryPath is the path used to query the application for the
+	// rows, defined by start row and end row, to data root merkle inclusion proof via
+	// the ABCI query method.
+	RowsInclusionProofQueryPath = "custom/rowsInclusionProof/%d/%d"
+	// RowsInclusionProofBySharesQueryPath is the path used to query the application for the
+	// rows, defined by a set of shares, to data root merkle inclusion proof via
+	// the ABCI query method.
+	RowsInclusionProofBySharesQueryPath = "custom/rowsInclusionProofByShares/%d/%d"
```

Comment:
I'm confused over the necessity to have 5 new queries. Want to discuss this synchronously?

Reply:
Sure, as an overview (for other reviewers):
- `TxSharesQueryPath`: takes a transaction index and returns the shares range which the transaction takes. This is useful to get the rows proof.
- `MessageSharesQueryPath`: takes a transaction index and returns the shares range which the message, that the transaction committed to, occupies.
- `ShareInclusionProofQueryPath`: NMT inclusion proof of shares to rows
- `RowsInclusionProofQueryPath`: Merkle inclusion proof for rows to data root, defined by a range of rows.
- `RowsInclusionProofBySharesQueryPath`: merkle inclusion proof for rows to data root, defined by a range of shares. This one is useful, since we don't have access to the rows indexes in the shares NMT proof.

`RowsInclusionProofQueryPath` will not be used, I guess, for the QGB. But, I thought it is good to provide that API too, as it might be useful to just prove certain rows to data root.

Reply:
Check now please when you have time, refactored to 1 query containing everything